Mathlib/Algebra/Polynomial/Derivation.lean

```diff
@@ -5,6 +5,7 @@ Authors: Kevin Buzzard, Richard Hill
 -/
 import Mathlib.Algebra.Polynomial.AlgebraMap
 import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Algebra.Polynomial.Module.Basic
```

Comment:
I think you should create a new file for this material. One shouldn't need to know about `PolynomialModule` just to talk about the basic material here.

Reply:
I don't really understand this concern. `PolynomialModule` is a basic file and the imports of that file are already imported by `Polynomial/Derivation`. Anyways splitting is still nice but I think it's confusing to have `Polynomial/Derivation` and `Derivation/Polynomial`. Can you give it a better name?

Reply:
Is there anything you suggest?

Reply:
Not sure. Maybe `Polynomial/Derivation/MapCoeffs` and the original file to `Polynomial/Derivation/Basic`

Reply:
I'm not sure if `Polynomial/Derivation` should be a directory, is there anything else which could be placed there?

Reply:
Perhaps the original file could be moved to `RingTheory/Derivation/Polynomial` and this will be `RingTheory/Derivation/MapCoeffs`?